onadata/apps/api/viewsets/dataview_viewset.py

```diff
@@ -77,25 +77,47 @@ def get_field_lookup(column, filter_string):
     return "json__" + column + filter_to_field_lookup(filter_string)
 
 
-def apply_filters(instance_qs, filters):
+def get_filter_kwargs(filters):
```

Comment:
I'm a bit against changing function signatures; Not sure of the potential impact of doing so

Reply:
There's only one place the `apply_filters` function is used (I have ensured it's updated to work with the new signature)

This change reduced the number of things `apply_filters` does (Gives it a single responsibility)

Reply:
I have also made sure that these changes are well tested